### DISCUSSION TITLE
Uploading glb/usdz file to use with model viewer

### DISCUSSION THREAD

**Discussion by rianlukog** [google/model-viewer#2480] [Q&A]
**This is my first post on GitHub, I have always been terrified of using it because I always considered it to be a place for only the smartest of the smartest, but here goes, 

I want to add a 3d model viewer + AR onto my wix site, I just cant seem to figure out how to "upload my glb/usdz file"**

**Comment by ThorstenBux** [comment]
My question would be what you mean with how to upload your glb 😀.
You can upload it to one publicly available web store that you have AWS S3, Google cloud store to name some popular ones. Then you take the public URL and use it when you add your model-viewer instance. There is nothing WIX specific there afaik.
Maybe you can post your code snippet of me to better understand.

**Reply by rianlukog** [reply]
Thanks for your reply,
https://imgur.com/a/UYNeqxy

What I want to achieve is something like the above ( Which I was able to do myself by using Vectary.com, where I uploaded a GLB/USDZ file and it returned an embed code which I then placed in the Wix HTML widget) 
Problem is that this service is paid and I was wondering if there is another way to do this. 

CODE: <iframe id="d47af27f-7536-4232-ad44-4a5a5693c6bf" src="https://www.vectary.com/viewer/v1/?model=d47af27f-7536-4232-ad44-4a5a5693c6bf&env=studio3" frameborder="0" width="100%" height="480"></iframe>

I suppose currently the GLB file is being read from vectary server

**Reply by rianlukog** [reply]
Update#1 
I have uploaded my GLB file to a "bucket" in Google cloud, and found the link for its location, I then added that to the model viewer src 

<model-viewer src="https://storage.cloud.google.com/productlibrarygk/T1.glb" camera-controls>

    <div class="progress-bar hide" slot="progress-bar">
        <div class="update-bar"></div>
    </div>
    <button slot="ar-button" id="ar-button">
        View in your space
    </button>
    <div id="ar-prompt">
        <img src="https://modelviewer.dev/shared-assets/icons/hand.png">
    </div>
</model-viewer>


My wix widget shows nothing except a hand icon holding a phone and a scrollbar

**Reply by rianlukog** [reply]
# Update 2 
Found an issue with the previous bucket setting where individual objects could not be accessed, I have changed it to public access, yet my model viewer issue persists. 

https://storage.googleapis.com/gkn/T1.glb

**Comment by rianlukog** [comment]
# update 3 

![image](https://user-images.githubusercontent.com/85927224/122066612-8587cc00-ce10-11eb-974a-6784ea63a383.png)

**Reply by elalish** [reply]
Have you checked the console? There are usually logs about things like 404 or CORS header problems. I personally would bet on the latter (this is a setting on the server).

**Reply by rianlukog** [reply]
I am very new to this, but thanks for your advice, there seems to be many confusing parts, I used a working example from vectary code ( as above shown vase example) and replace only the 'src' with my google cloud link, however I still do not know what to do with id="d47a..... etc etc"

**Reply by ThorstenBux** [reply]
Yeah, most likely CORS:
```
Access to XMLHttpRequest at 'https://storage.googleapis.com/gkn/T1.glb' from origin 'https://192.168.1.134:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

https://cloud.google.com/storage/docs/cross-origin

**Reply by rianlukog** [reply]
# UPDATE I GOT IT TO WORK by using glitch.me, uploaded the files (GLB,USDZ) to it and ran the below code and embedded it in wix, only missing this is the AR QR code button on web

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Model Viewer</title>
    <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
    <style>
model-viewer{
    width: 400px;
    height: 600px;
   

}
    </style>
</head>
<body>

<model-viewer src="https://cdn.glitch.com/d82e92a7-03d8-4967-9684-9b99a4eca91b%2FT1.glb?v=1623787635650" camera-controls shadow-intensity="1" ar ar-modes="webxr scene-viewer quick-look">

    <div class="progress-bar hide" slot="progress-bar">
        <div class="update-bar"></div>
    </div>
    <button slot="ar-button" id="ar-button">
        View in your space
    </button>
    <div id="ar-prompt">
        <img src="https://modelviewer.dev/shared-assets/icons/hand.png">
    </div>
</model-viewer>

</html>

![image](https://user-images.githubusercontent.com/85927224/122207247-8e83a680-cebf-11eb-9d77-8bb614488af5.png)
![image](https://user-images.githubusercontent.com/85927224/122207278-95aab480-cebf-11eb-9b35-652b343adb9c.png)

--------------------------------------------

How can I embed the QR generation button ?

**Reply by Billelouazene** [reply]
hi ! 
im want to do exactly the same thing, ad a 3d object on wix with user interaction,
but can you be more specific on how you have done it with glitch please ? step by step ?
It would be very helpful !! 

thank you so much.

**Comment by rianlukog** [comment]
![image](https://user-images.githubusercontent.com/85927224/122076313-65f4a180-ce18-11eb-8d06-9229b494ba45.png)

**Reply by elalish** [reply]
I have no idea what you mean by QR generation, but if you want AR, you should put the `ar` attribute in. See the docs: https://modelviewer.dev/docs/index.html#augmentedreality-attributes
